pkg/datapath/linux/devices_controller.go

```diff
@@ -362,7 +362,8 @@ func (dc *devicesController) processUpdates(
 
 func deviceAddressFromAddrUpdate(upd netlink.AddrUpdate) tables.DeviceAddress {
 	return tables.DeviceAddress{
-		Addr: ip.MustAddrFromIP(upd.LinkAddress.IP),
+		Addr:      ip.MustAddrFromIP(upd.LinkAddress.IP),
+		Secondary: upd.Flags&unix.IFA_F_SECONDARY == 0,
```

Comment:
Ah had fixed this already in the original branch I forked this from, but looks like that change didn't carry over (this happened because this was originally `Primary` and didn't fix this line). Fixing this again and making sure to have the test coverage for it.